user_secret.c

```diff
@@ -0,0 +1,230 @@
+#include "user_secret.h"
+
+// use mock functions when build for UT
+#if defined (UNIT_TEST)
+void *mock_malloc(size_t size);
+void mock_free(void* ptr);
+#define malloc  mock_malloc
+#define free    mock_free
+#else
+#endif
+
+/* Macros for have_next_line result */
+#define HAVE_NEXT_SETTING_LINE 1
+#define NO_NEXT_SETTING_LINE   0
+
+/* Macros for parse user input */
+#define USER_COMMAND_TOKEN_WHITESPACE              " \t\n\r\f"
+#define USER_COMMAND_TOKEN_SETTING_SPLITTER        " =\t"
+#define USER_COMMAND_TOKEN_EQUAL                   "="
+#define USER_COMMAND_TOKEN_COMMA                   ","
+
+/* The command alias prefix */
+const char* COMMAND_ALIAS = "Cmnd_Alias";
+
+/* The user secret setting */
+const char* USER_SECRET_SETTING = "PASSWD_CMDS";
+
+/* Regex list */
+REGEX_NODE *global_regex_list = NULL;
+
+/* Append regex to list */
+int append_regex_to_list(regex_t regex)
+{
+    /* Create and initialize regex node */
+    REGEX_NODE *new_regex_node = (REGEX_NODE *)malloc(sizeof(REGEX_NODE));
+    if (new_regex_node == NULL)
+    {
+        /* When allocate memory failed, stop and return. also output log to both syslog and stderr with LOG_PERROR*/
+        trace("Failed to allocate memory for regex node.\n");
+        return REGEX_APPEND_FAILED;
+    }
+
+    new_regex_node->next = NULL;
+    new_regex_node->regex = regex;
+
+    /* Find the pointer to the latest regex node's 'next' field */
+    REGEX_NODE **current_node = &global_regex_list;
+    while (*current_node != NULL) {
+        current_node = &((*current_node)->next);
+    }
+    
+    /* append new regex to tail node */
+    *current_node = new_regex_node;
+    return REGEX_APPEND_SUCCESS;
+}
+
+/* Release user secret setting */
+void release_user_secret_setting()
+{
+    if (global_regex_list == NULL) {
+        return;
+    }
+
+    /* Walk to last regex */
+    REGEX_NODE *next_node = global_regex_list;
+    while (next_node != NULL) {
+        /* Continue with next regex */
+        REGEX_NODE* current_node_memory = next_node;
+        next_node = next_node->next;
+        
+        /* Free node memory, this may also reset all allocated memory depends on c lib implementation */
+        free(current_node_memory);
+    }
+
+    /* Reset list */
+    global_regex_list = NULL;
+}
+
+/* Replace user secret with regex */
+char * remove_user_secret(const char* command)
+{
+    if (global_regex_list == NULL) {
+        return NULL;
+    }
+
+    /* Check every regex */
+    REGEX_NODE *next_node = global_regex_list;
+    while (next_node != NULL) {
+        /* Try fix user secret with current regex */
+        char* fixed_command = remove_user_secret_by_regex(command, next_node->regex);
+        if (fixed_command != NULL) {
+            return fixed_command;
+        }
+        
+        /* If user secret not fix, continue try next regex */
+        next_node = next_node->next;
+    }
+
+    return NULL;
+}
+
+/* Trim start */
+char* trim_start(char *str)

```

Comment:
>trim_start [](http://example.com/codeflow?start=6&length=10)

`trim` is indicating in-place modification. How about `find_non_space` ?

Reply:
Fixed.

Reply:
I see see `trim`

Reply:
Fixed, now rename to find_non_space.